### ISSUE TITLE
possible to ban keywords?

### ISSUE THREAD

**Issue by AtariBaby** [evilhero/mylar#2095]
Hi is it possible to reject files if they contain keywords? I recently downloaded a file with a ".info" url in the filename, which broke a rule in some channels where I share comics.

**Comment by evilhero** [comment]
Mylar only looks for cbr/cbz files, but did you download the file using mylar? 

Was the url within the cbr/cbz file itself?

Can you indicate what the naming was of said file?

**Comment by AtariBaby** [comment]
Hi I am pretty sure it was dl'd through mylar: Injection 014 (2017) GetComics.INFO
I haven't even read that series yet, so I don't think I manually got it from anywhere.
Sonarr has a feature where you can prefer files or block files based on filename keywords. That's why I asked. If it's not possible on the front end or back end, I'll just have to be vigilant! It only occurred once so far, so not a huge deal.

**Comment by evilhero** [comment]
That wouldn't have been downloaded thru mylar - most likey it was from from the url in the filename as they 'tag' all their files with that (it was in a zip which contained the cbr/cbz and the .INFO file). I have never seen one of their items on newsgroups, or torrents (public maybe altho haven't checked either). 

When post-processing files Mylar will only copy/move/delete cbr/cbz files and leaves everything else as is - mainly due to people just dumping thru comics in one directory which has a mish-mosh of different files.

So in this case because the .info file was part of a zip which contained it and the cbr file, there's not much that can be done from within Mylar.   Unless you want to make use of the pre-post-processing script option and make something to address the problem prior to post-process occuring.

**Comment by evilhero** [comment]
It looks like that's available on dog, so odds are it's on other indexers as well (thnx for the info bbq).

The parser should be able to pick it up, recognize it and rename it so that it's 'normalized.'  However if you don't rename files, which I'm guessing is plausible since you're sharing with others, you'd want the original file name intact. And since these files don't contain the scanner's tag, they're probably not legitimately named anyways.

Long story short, I dunno - just more info to dissect at this point ;)

**Comment by AtariBaby** [comment]
Yeah in my case it’s important to preserve the original file. Hi Cher comics in “no repacks“ circles, and that would include not re-naming the file. I can just try to remember to be vigilant if there’s no way to tell Mylar to skip some file names 

--
Gregory Fleming
415-218-4101
Misspelled from my iPhone

**Comment by leaderdog** [comment]
With some of those sites, they repack them with the comic website readme or link inside which breaks the original hash.  When you go into dc++ I had a bunch of files flagged it took me some time to figure out which ones were doing it, and most were the ones I grabbed off sites like that to fill missing books in the run.

Obviously, if that's not a concern, then it's no big deal, but for dc++ it's a whole process of files needed to be run to check the hashes and blargh, it's a bit annoying. ;)

**Comment by AtariBaby** [comment]
I wish I was more of a programmer. I’d love to develop some kind of engine that bridges Mylar to to dc++

As it is I try to be very careful and like you said, grab missing issues and older comics through the pure sources, and only use Mylar for new releases 

--
Gregory Fleming
415-218-4101
Misspelled from my iPhone
